Mathlib/Order/Nucleus.lean

```diff
@@ -0,0 +1,44 @@
+/-
+Copyright (c) 2024 Christian Krause. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chriara Cimino, Christian Krause
+-/
+import Mathlib.Order.CompleteBooleanAlgebra
+
+/-!
+# Nucleus
+Locales are the dual concept to Frames.
+A Nucleus is a function between Frames which corresponds to a sublocale.
+https://ncatlab.org/nlab/show/nucleus
+-/
+variable {X : Type*} [Order.Frame X]
+
+/--
+The Type of Nuclei on a Frame.
+-/
+structure Nucleus (X : Type*) [Order.Frame X] where
```

Comment:
Bundled morphisms in Mathlib should come with `FunLike` instance, and possibly with `NucleusClass`. See any other bundled morphism (e.g., `InfHom` which you may want to extend) for an example.